cartography/intel/spacelift/ec2_ownership.py

```diff
@@ -0,0 +1,194 @@
+import json
+import logging
+import re
+from typing import Any
+
+import boto3
+import neo4j
+
+from cartography.client.core.tx import load_matchlinks
+from cartography.models.spacelift.run import SpaceliftRunToEC2InstanceRel
+from cartography.util import aws_handle_regions
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+# Regex pattern to match EC2 instance IDs
+INSTANCE_ID_PATTERN = re.compile(r"\b(i-[0-9a-f]{8,17})\b")
+
+
+@aws_handle_regions
+@timeit
+def get_ec2_ownership(
+    aws_session: boto3.Session, bucket_name: str, object_key: str
+) -> list[dict[str, Any]]:
+    """
+    Fetch EC2 ownership data from S3 bucket containing Athena query results.
+    """
+    logger.info(f"Fetching EC2 ownership data from s3://{bucket_name}/{object_key}")
+
+    # Create S3 client from the boto3 session
+    s3_client = aws_session.client("s3")
+
+    # Get the object from S3
+    response = s3_client.get_object(Bucket=bucket_name, Key=object_key)
+
+    # Read the object body (returns a StreamingBody)
+    object_body = response["Body"].read()
+
+    # Decode bytes to string and parse JSON
+    json_content = json.loads(object_body.decode("utf-8"))
```

Comment:
`json_content = json.load(io.TextIOWrapper(response["Body"], encoding="utf-8"))`
Something like this may be more efficient for large files, just something to think about if you run into sluggishness, non blocker